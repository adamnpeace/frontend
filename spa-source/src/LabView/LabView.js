import React, { Component } from "react";
import brace from "brace";
import marked from "marked";
import { Link } from "react-router-dom";
import AceEditor from "react-ace";
import { Row, Col, Button } from "reactstrap";
import Footer from "../Footer/Footer";
import styles from "./LabView.module.css";
import "brace/mode/python";
import "brace/theme/monokai";
import "brace/theme/textmate";

import axios from "axios";

import createHistory from "history/createBrowserHistory";

class LabView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      content: " ",
      code: " "
    };
  }

  componentDidMount() {
    const ip = `http://127.0.0.1:8000/api/`;
    axios
      .get(ip + /*this.props.labID + `/`*/ +this.props.classID + `/`)
      .then(res => {
        const content = unescape(res.data.content);
        const code = res.data.name;
        this.setState({ code });
        this.setState({ content });
      });
  }

  render() {
    const history = createHistory();
    const style = {
      fontSize: "14px !important",
      border: "1px solid lightgray",
      width: "100%",
      height: "100%"
    };
    return (
      <div>
        <div>
          <Row>
            <Col xs={6} md={6}>
              <article
                dangerouslySetInnerHTML={{ __html: marked(this.state.content) }}
              />
            </Col>
            <Col xs={6} md={6}>
              <AceEditor
                style={style}
                mode="python"
                theme="textmate"
                name="blah2"
                onLoad={this.onLoad}
                onChange={this.onChange}
                fontSize={14}
                showPrintMargin={true}
                showGutter={true}
                highlightActiveLine={true}
                value={this.state.code}
                setOptions={{
                  enableBasicAutocompletion: false,
                  enableLiveAutocompletion: false,
                  enableSnippets: false,
                  showLineNumbers: true,
                  tabSize: 2
                }}
              />
            </Col>
          </Row>
        </div>
        <nav class="navbar fixed-bottom navbar-expand-md navbar-light bg-light">
          <Link to="/" class="nav-link">
            <button type="button" class="btn btn-primary">
              Back
            </button>
          </Link>
          <a
            href={`/editor/${this.props.labID}/${parseInt(this.props.classID) +
              1}`}
            class="nav-link"
          >
            <button type="button" class="btn btn-primary">
              Next
            </button>
          </a>
        </nav>
      </div>
    );
  }
}

export default LabView;
