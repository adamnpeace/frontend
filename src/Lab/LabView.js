import React from "react";
import axios from "axios";
import {Box, Button, Heading, Markdown} from "grommet";
import Editor from "./Editor";
import loading from "../img/loading.svg";
import * as Icons from "grommet-icons";
import {Config} from "../Config";
import {withRouter} from "react-router";

class LabView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
    };
  }
  
  goTo = route => {
    this.props.history.push(route);
  };
  
  render() {
    if (typeof (this.props.lesson) === "string") {
      return (
        
        <div>
          <Box
            height="83vh"
            direction="row"
            border={{
              color: "accent-3",
              size: "large"
            }}
            pad="medium"
            elevation="medium"
          >
            <Box pad="small" basis="2/4" overflow="auto" elevation="xsmall">
              <Heading>{this.props.lesson}</Heading>
            
            </Box>
            
            <Box pad="small" basis="2/4">
              <Editor code={this.props.lesson}/>
            </Box>
          </Box>
        </div>
      )
    } else {
      return (
        <img
          src={loading}
          alt="..."
          style={{position: "absolute", top: "30%", left: "48%"}}
        />)
    }
  }
}

export default LabView;
