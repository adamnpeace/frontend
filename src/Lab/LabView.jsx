import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { Box, Heading } from 'grommet';
import Markdown from 'markdown-to-jsx';
import Editor from './Editor';
import loading from '../img/loading.svg';

const propTypes = {
  changeCode: PropTypes.func.isRequired,
  lessonID: PropTypes.number.isRequired,
  lesson: PropTypes.object.isRequired,
  codeSnippet: PropTypes.string.isRequired
};
const defaultProps = {};

class LabView extends React.Component {
  onChangeCode(newCode) {
    this.props.changeCode(newCode, this.props.lessonID);
  }

  render() {
    const { codeSnippet, lesson } = this.props;

    if (typeof this.props.lesson.name === 'string') {
      return (
        <>
          <Box pad="small" basis="2/4" overflow="auto" elevation="xsmall">
            <Heading>{lesson.name}</Heading>
            <Markdown style={{ 'font-family': 'Lato' }}>{lesson.content}</Markdown>
          </Box>
          <Box pad="small" basis="2/4">
            <Editor codeSnippet={codeSnippet} changeCode={this.onChangeCode.bind(this)} />
          </Box>
        </>
      );
    } else {
      return (
        <img src={loading} alt="..." style={{ position: 'absolute', top: '30%', left: '48%' }} />
      );
    }
  }
}

LabView.propTypes = propTypes;
LabView.defaultProps = defaultProps;

export default withRouter(LabView);
