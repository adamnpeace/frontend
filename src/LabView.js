import React from "react";
import axios from "axios";
import { Box, Button, Grid, Heading, Markdown, Text } from "grommet";
import AceEditor from "react-ace";
import ShowContent from "./ShowContent";
import "brace/theme/textmate";
import Editor from "./Editor";
import { isNullOrUndefined } from "util";
import marked from "marked";
import loading from "./loading.svg";
import { timeout } from "q";

class LabView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      num: 8,
      snippet: "",
      returnValue: "",
      content: "",
      name: "",
      defaultCode: ""
    };
    this.apiUrl = "http://52.151.66.189/api/";
  }

  sleep(time) {
    return new Promise(resolve => setTimeout(resolve, time));
  }

  componentDidMount() {
    this.setState({ isLoading: true });
    axios.get(this.apiUrl + this.state.num).then(res => {
      this.sleep(100).then(() => {
        this.setState({
          content: res.data.content,
          name: res.data.name,
          defaultCode: res.data.code,
          isLoading: false
        });
      });
    });
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state.num !== prevState.num) {
      this.setState({ isLoading: true });
      axios.get(this.apiUrl + this.state.num).then(res => {
        this.sleep(300).then(() => {
          this.setState({
            content: res.data.content,
            name: res.data.name,
            defaultCode: res.data.code,
            isLoading: false
          });
        });
      });
    }
    if (this.state.snippet !== prevState.snippet) {
      console.log("Computing " + this.state.snippet);
      axios
        .get(this.apiUrl + "test/", {
          params: { model_input: this.state.snippet }
        })
        .then(res => {
          this.setState({ returnValue: res.data.complex_result });
        });
    }
  }

  computeOutput(snippet) {
    this.setState({ snippet: snippet });
  }

  handlePrev() {
    var newNum = this.state.num - 1;
    this.setState({ num: newNum });
  }

  handleNext() {
    var newNum = this.state.num + 1;
    this.setState({ num: newNum });
  }

  render() {
    return (
      <div>
        {!this.state.isLoading ? (
          <div>
            <Heading>{this.state.name}</Heading>
            <br />
            <Markdown>{this.state.content}</Markdown>

            <Editor
              defaultCode={this.state.defaultCode}
              computeOutput={this.computeOutput.bind(this)}
            />
            {this.state.returnValue !== "" ? (
              <div>
                <h2>Result:</h2>
                <br />
                {this.state.returnValue}
              </div>
            ) : (
              <div />
            )}
            <br />
            <Box
              tag="footer"
              direction="row"
              justify="end"
              pad="medium"
              gap="small"
              flex={false}
              background="brand"
            >
              <Button label="Back" onClick={this.handlePrev.bind(this)} />
              <Button label="Next" onClick={this.handleNext.bind(this)} />
            </Box>
          </div>
        ) : (
          <img
            src={loading}
            alt="..."
            style={{ position: "absolute", top: "30vh", left: "50vw" }}
          />
        )}
      </div>
    );
  }
}
export default LabView;
