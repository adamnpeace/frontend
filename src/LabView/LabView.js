import React from "react";
import axios from "axios";
import { Box, Button, Heading, Markdown } from "grommet";
import Editor from "./Editor";
import loading from "../loading.svg";
import * as Icons from "grommet-icons";

class LabView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      num: this.props.classID,
      content: "",
      name: "",
      defaultCode: ""
    };
    this.apiUrl = "http://52.151.66.189/api/";
  }

  sleep(time) {
    return new Promise(resolve => setTimeout(resolve, time));
  }

  componentDidMount() {
    this.setState({ isLoading: true });
    axios.get(this.apiUrl + this.state.num).then(res => {
      this.sleep(100).then(() => {
        this.setState({
          content: res.data.content,
          name: res.data.name,
          defaultCode: res.data.code,
          isLoading: false
        });
      });
    });
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state.num !== prevState.num) {
      this.setState({ isLoading: true });
      axios.get(this.apiUrl + this.state.num).then(res => {
        this.sleep(300).then(() => {
          this.setState({
            content: res.data.content,
            name: res.data.name,
            defaultCode: res.data.code,
            isLoading: false
          });
        });
      });
    }
  }

  handlePrev() {
    var newNum = this.state.num - 1;
    this.setState({ num: newNum });
  }

  handleNext() {
    var newNum = this.state.num + 1;
    this.setState({ num: newNum });
  }

  render() {
    return (
      <div>
        {!this.state.isLoading ? (
          <Box
            height="83vh"
            direction="row"
            border={{
              color: "accent-3",
              size: "large"
            }}
            pad="medium"
          >
            <Box pad="small" basis="2/4" overflow="auto" elevation="xsmall">
              <Heading>{this.state.name}</Heading>
              <Markdown>{this.state.content}</Markdown>
            </Box>

            <Box pad="small" basis="2/4">
              <Editor defaultCode={this.state.defaultCode} />
            </Box>
          </Box>
        ) : (
          <Box
            height="83vh"
            direction="row"
            border={{
              color: "accent-3",
              size: "large"
            }}
            pad="medium"
          >
            <img
              src={loading}
              alt="..."
              style={{ position: "absolute", top: "30%", left: "48%" }}
            />
          </Box>
        )}

        <Box
          tag="footer"
          border={{
            side: "bottom",
            color: "accent-3",
            size: "large"
          }}
          direction="row"
          elevation="large"
          height="60px"
          justify="end"
          pad="xsmall"
          gap="small"
          flex={false}
          background="accent-3"
        >
          <Button
            icon={<Icons.FormPrevious />}
            label="Back"
            onClick={this.handlePrev.bind(this)}
            primary
          />
          <Button
            label="Next"
            icon={<Icons.FormNext />}
            margin={{ right: "10px" }}
            onClick={this.handleNext.bind(this)}
            primary
            reverse
          />
        </Box>
      </div>
    );
  }
}
export default LabView;
