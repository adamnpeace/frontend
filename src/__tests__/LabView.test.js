import React from "react";
import ReactDOM from "react-dom";
import LabView from '../LabView';
import { StaticRouter } from 'react-router';
import { Link } from 'react-router-dom';
import renderer from "react-test-renderer";

describe("Entire App Test", () => {
  const context = {};

  it("LabView app renders without crashing", () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <StaticRouter location="location" context={context}>
        <LabView />
      </StaticRouter>, div);
  });

  it("LabView renders correctly according to first snapshot", () => {
    const component = renderer.create(
      <StaticRouter location="location" context={context}>
        <LabView />
      </StaticRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
