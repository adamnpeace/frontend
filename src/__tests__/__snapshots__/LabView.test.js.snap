// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entire App Test LabView renders correctly according to first snapshot 1`] = `
<div>
  <div
    className="StyledBox-sc-13pk1d4-0 yuWhC"
  >
    <img
      alt="..."
      src="loading.svg"
      style={
        Object {
          "left": "48%",
          "position": "absolute",
          "top": "30%",
        }
      }
    />
  </div>
  <footer
    className="StyledBox-sc-13pk1d4-0 gQsyIH"
  >
    <button
      className="StyledButton-sc-323bzc-0 kvjDlM"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="button"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 hHwlHq"
      >
        <svg
          aria-label="FormPrevious"
          className="StyledIcon-ofa7kd-0 jAaLrm"
          viewBox="0 0 24 24"
        >
          <polyline
            fill="none"
            points="9 6 15 12 9 18"
            stroke="#000"
            strokeWidth="2"
            transform="matrix(-1 0 0 1 24 0)"
          />
        </svg>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
        />
        Back
      </div>
    </button>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
    />
    <button
      className="StyledButton-sc-323bzc-0 dUvNsR"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="button"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 hHwlHq"
      >
        Next
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
        />
        <svg
          aria-label="FormNext"
          className="StyledIcon-ofa7kd-0 jAaLrm"
          viewBox="0 0 24 24"
        >
          <polyline
            fill="none"
            points="9 6 15 12 9 18"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
      </div>
    </button>
  </footer>
</div>
`;
